### PR TITLE
Fix VS Code extension to resolve file paths relative to treatment file

### LINKED ISSUE #120: VS Code extension resolves file paths relative to workspace root, not treatment file
## Problem

We made a breaking change to resolve prompt file paths (and other resource paths) relative to the treatment file where they are referenced. The runtime resolution in the viewer and StagebookProvider contract both implement this correctly. However, the VS Code extension still resolves `file:` paths relative to the workspace root.

This means:
- **False "File not found" diagnostics** — if `study/treatment.treatments.yaml` references `file: prompts/consent.prompt.md`, the extension checks for `<workspace>/prompts/consent.prompt.md` instead of `<workspace>/study/prompts/consent.prompt.md`.
- **Wrong quick-fix suggestions** — the "Did you mean?" paths are workspace-root-relative, so accepting one inserts a path that doesn't match the runtime convention.
- **Wrong autocomplete paths** — completions are also workspace-root-relative.
- **Stage preview can't load files** — the webview's `readFile` handler and `webviewBaseUri` both resolve against the workspace root.

## Affected code

All in `apps/vscode/src/extension.ts`:

| Feature | Lines | What's wrong |
|---|---|---|
| `checkFileReferences()` | 148–152 | `vscode.Uri.joinPath(workspaceFolder.uri, filePath)` — resolves against workspace root |
| `FilePathQuickFixProvider` | 323–328 | `path.relative(workspaceFolder.uri.fsPath, ...)` — suggestions are workspace-root-relative |
| `FilePathCompletionProvider` | 392–396 | `path.relative(workspaceFolder.uri.fsPath, ...)` — completions are workspace-root-relative |
| Stage preview `readFile` handler | 757–760 | `vscode.Uri.joinPath(currentWorkspaceFolder.uri, filePath)` — resolves against workspace root |
| Stage preview `webviewBaseUri` | 735–737 | Sends workspace root URI as base |

## Proposed fix

Derive a **treatment file directory URI** from `document.uri` and use that as the base for all path operations:

```typescript
const treatmentDir = vscode.Uri.joinPath(document.uri, "..");
```

Concretely:

1. **`checkFileReferences`**: Change `vscode.Uri.joinPath(workspaceFolder.uri, filePath)` to `vscode.Uri.joinPath(treatmentDir, filePath)`. Keep the existing workspace-root traversal guard.

2. **`FilePathQuickFixProvider`**: Compute `path.relative(treatmentDirFsPath, f.fsPath)` instead of relative to workspace root.

3. **`FilePathCompletionProvider`**: Same — compute relative paths from the treatment file's directory.

4. **Stage preview**: Derive `webviewBaseUri` and `readFile` base from the treatment file's location, not the workspace folder. The treatment file URI needs to be captured when the preview command runs and threaded through to the message handler.

### PR DESCRIPTION
**Pull request by JamesPHoughton**
## Summary

- **Fix path resolution**: All `file:` path lookups (existence checks, quick-fix suggestions, autocomplete completions, stage preview) now resolve relative to the treatment file's directory instead of the workspace root.
- **Harden readFile handler**: Add post-resolution workspace boundary check to the stage preview's `readFile` handler (defense in depth against symlinks).
- **Extract testable helpers**: `isWithinWorkspace` and `relativizePath` in `lib/filePaths.ts` with 11 unit tests, consistent with the existing pattern of pure `lib/` modules.
- **Fix stale quick-fix cache**: Invalidate cached suggestions when switching between treatment files in different directories.
- **Clean up dead code**: Replace mutable `currentWorkspaceFolder` with a local const + stored `fsPath` string.

## Test plan

- [x] All 131 vscode extension tests pass (120 existing + 11 new)
- [x] All 469 stagebook library tests pass
- [x] All 60 viewer tests pass
- [ ] Manual: open a treatment file in a subdirectory, verify "File not found" diagnostics resolve relative to that subdirectory
- [ ] Manual: verify autocomplete and quick-fix suggestions show paths relative to the treatment file

Fixes #120

https://claude.ai/code/session_017pPtpx1ESXVoQQieuzFsgM

### CODE REVIEW

File: apps/vscode/src/lib/filePaths.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Returns true if `resolvedFsPath` is exactly equal to `workspaceRootFsPath`
+ * or is a descendant of it.
+ */
+export function isWithinWorkspace(
+  resolvedFsPath: string,
+  workspaceRootFsPath: string,
+): boolean {
+  if (resolvedFsPath === workspaceRootFsPath) return true;
+  return resolvedFsPath.startsWith(workspaceRootFsPath + path.sep);
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/lib/filePaths.ts]: Good catch — both edge cases were real. Rewrote using `path.relative` in 9dd261c so trailing separators and `"/"` as a workspace root are handled correctly. Added tests for both cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/vscode/src/lib/filePaths.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+describe("isWithinWorkspace", () => {
+  it("returns true when resolved path equals workspace root", () => {
+    expect(isWithinWorkspace("/workspace", "/workspace")).toBe(true);
+  });
+
+  it("returns true for a direct child", () => {
+    expect(
+      isWithinWorkspace(path.join("/workspace", "file.txt"), "/workspace"),
+    ).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/lib/filePaths.test.ts:16]: You're right — switched to `path.join(path.sep, ...)` throughout in 9dd261c so the tests use the platform's native separator for both roots and joined paths. Also added explicit tests for filesystem root and trailing-separator cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/vscode/src/extension.ts [hunk 3]
```diff
@@ -755,9 +757,22 @@ export function activate(context: vscode.ExtensionContext): void {
             }
             try {
               const fileUri = vscode.Uri.joinPath(
-                currentWorkspaceFolder.uri,
+                currentTreatmentDir,
                 filePath,
               );
+              // Post-resolution workspace boundary check (defense in depth
+              // against symlinks and edge cases the substring check misses)
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Correct — the check is a string comparison on `fsPath`, not a `realpath` resolve, so a symlink inside the workspace pointing out would still be followed by `fs.readFile`. Updated the comment in 9dd261c to accurately describe what the guard does (path traversal + shared-prefix edge cases) and explicitly call out the symlink gap. A real-path defense would need an async `realpath` call; happy to add that as a follow-up if it's worth the latency.

---
_Generated by [Claude Code](https://claude.ai/code)_
